x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/RolloverStep.java

```diff
@@ -115,10 +127,18 @@ public void performAction(
             }
 
             rolloverTarget = rolloverAlias;
+            targetFailureStore = false;
         }
 
         // Calling rollover with no conditions will always roll over the index
         RolloverRequest rolloverRequest = new RolloverRequest(rolloverTarget, null).masterNodeTimeout(TimeValue.MAX_VALUE);
+        if (targetFailureStore) {
+            rolloverRequest.setIndicesOptions(
+                IndicesOptions.builder(rolloverRequest.indicesOptions())
+                    .failureStoreOptions(opts -> opts.includeFailureIndices(true).includeRegularIndices(false))
```

Comment:
I'm not sure I get the value/reason of doing this over `IndicesOptions.FailureStoreOptions.builder().includeRegularIndices(true).includeFailureIndices(true)`? Although, I could see value in adding some static fields on `FailureStoreOptions` for its permutations.

Reply:
The reason for this was two fold:
1. If we ever add a new field on FailureStoreOptions, then every place that we modify the settings will need to be changed to copy the existing FailureStoreOptions. This is a smaller kind of the problem that led to the big refactoring of IndicesOptions - adding things to it was getting invasive.
2. Fewer keystrokes than doing `rolloverRequest.setIndicesOptions(IndicesOptions.builder(rolloverRequest.indicesOptions()).failureStoreOptions(IndicesOptions.FailureStoreOptions.builder(rolloverRequest.indicesOptions().failureStoreOptions()).includeFailureIndices(true).includeRegularIndices(false)).build()`

Reply:
Ah! Second point indeed wins :)